src/core/RenderContext.cpp

```diff
@@ -130,17 +140,21 @@ void RenderContext::updateGLWindows()
 
     BOOST_FOREACH( WallWindowPtr window, windows_ )
     {
+        window->setBlockDrawCalls( false );
         window->viewport()->repaint();
+        window->setBlockDrawCalls( true );
         ++activeGLWindowIndex_;
     }
+    glFinish();
 }
 
 void RenderContext::swapBuffers()
 {
     BOOST_FOREACH( WallWindowPtr window, windows_ )
     {
-        static_cast<QGLWidget*>( window->viewport( ))->makeCurrent();
-        static_cast<QGLWidget*>( window->viewport( ))->swapBuffers();
+        QGLWidget* glContext = static_cast<QGLWidget*>( window->viewport( ));
+        glContext->makeCurrent();
+        glContext->swapBuffers();
```

Comment:
OK, I hadn't read the manpage correctly. I couldn't determine if this actually has an effect on the swapbuffer command though (intuitively I would expect that the swapbuffer should always be "flushed" as fast as possible), but since it can't hurt -> #167